src/v/v8_engine/script.cc

```diff
@@ -13,15 +13,25 @@
 #include "utils/file_io.h"
 
 #include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sleep.hh>
 #include <seastar/core/temporary_buffer.hh>
 #include <seastar/util/later.hh>
 
+#include <chrono>
+#include <functional>
 #include <string_view>
 #include <utility>
 
 namespace v8_engine {
 
-script::script(size_t max_heap_size_in_bytes) {
+script::script(
+  size_t max_heap_size_in_bytes,
+  v8_engine::executor_wrapper& executor_for_script,
```

Comment:
I wouldn't make executor to be part of a script. Maybe it would be better to have method like: 
```c++
script::run(executor&)
```
or 
```c++
auto task = script::get_task();
executor.run(task);
```

Reply:
fix